File: Frontend/SimSpot/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { OrderItem } from "./order-item";
import { Order } from "./order";

export class Purchase {
    customer!: Customer;
    shippingAddress!: Address;
    billingAddress!: Address;
    order!: Order;
    orderItems!: OrderItem[];
}

File: Backend/spring-boot-ecommerce/src/main/java/com/SimSpot/ecommerce/dto/Purchase.java
package com.SimSpot.ecommerce.dto;

import com.SimSpot.ecommerce.entity.Customer;
import com.SimSpot.ecommerce.entity.Address;
import com.SimSpot.ecommerce.entity.Order;
import com.SimSpot.ecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;

}
